clang/include/clang/AST/ExprCXX.h

```diff
@@ -4381,11 +4381,13 @@ class PackIndexingExpr final
 
   PackIndexingExpr(QualType Type, SourceLocation EllipsisLoc,
                    SourceLocation RSquareLoc, Expr *PackIdExpr, Expr *IndexExpr,
-                   ArrayRef<Expr *> SubstitutedExprs = {})
+                   ArrayRef<Expr *> SubstitutedExprs = {},
+                   bool EmptyPack = false)
       : Expr(PackIndexingExprClass, Type, VK_LValue, OK_Ordinary),
         EllipsisLoc(EllipsisLoc), RSquareLoc(RSquareLoc),
         SubExprs{PackIdExpr, IndexExpr},
-        TransformedExpressions(SubstitutedExprs.size()) {
+        TransformedExpressions(EmptyPack ? size_t(-1)
```

Comment:
I would much prefer we would do something like

```cpp
unsigned ExpandedToEmptyPack : 1;
size_t TransformedExpressions : sizeof(size_t) -1;
```
To avoid the magic value

Reply:
Thanks, that looks great! I almost forgot the bit fields things...Will do that tomorrow.